### ISSUE TITLE
Oops, something went wrong, no error code

### ISSUE THREAD

**Issue by KingKongst** [LedgerHQ/ledger-live-desktop#1989]
#### Ledger Live desktop, windows 7 64b

Hello, here is my problem, ledger live wont open, even after a re-install, the same problem
![error ledger](https://user-images.githubusercontent.com/50631545/57712791-7c483600-7671-11e9-99f9-34b9a22dcf45.jpg)

Also when i install ledger live to another pc, i am missing 2 accounts, stratis and bitcoin, when i open ledger on my mobile they are there, wtih the date of sending but when i connect my nano s to a new pc and i want to import an account, 2 of the 5 are gone......Help!

with kind regards, Douwe

**Comment by KingKongst** [comment]
SyntaxError: Unexpected token  in JSON at position 0
SyntaxError: Unexpected token  in JSON at position 0
    at JSON.parse (<anonymous>)
    at I (file:///C:/Program%20Files/Ledger%20Live/resources/app.asar/renderer.js:1:15561)
    at <anonymous>
redux failed: Invariant Violation: Could not find "store" in either the context or props of "Connect(G)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(G)".
Invariant Violation: Could not find "store" in either the context or props of "Connect(G)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(G)".
    at invariant (C:\Program Files\Ledger Live\resources\app.asar\node_modules\invariant\invariant.js:40:15)
    at new Connect (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-redux\lib\components\connectAdvanced.js:134:33)
    at vf (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:132:170)
    at Og (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:167:149)
    at Tg (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:180:30)
    at bi (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:232:22)
    at ci (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:233:284)
    at Di (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:249:348)
    at Yh (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:248:144)
    at Xh (C:\Program Files\Ledger Live\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:245:332)

**Comment by gre** [comment]
The error is strange, we'll need to reproduce it.
In the meantime to fix it, you will probably have to **Reset** and re-import the accounts.

**Comment by KingKongst** [comment]
thanks for responding, the reset button doesn't work
i have tried to re import my stratis account on the mobile app after removing it from the phone, now its totally gone.....

**Comment by gre** [comment]
@KingKongst please contact our ledger support and export the logs as soon as the error appear (don't send error on github, only on our tech support), you can attach this github issue and tell them to ping me (@gre)

thanks

**Comment by KingKongst** [comment]
i did last week, no respons.....
how can i export the logs? ledger live won;t open and i don't know how with ledger mobile

**Comment by gre** [comment]
can you tell me the support ticket number? mmh I think that's the button "Save" on that error interface. Does Ctrl+E works?

**Comment by KingKongst** [comment]
save button doesn't work, neither ctrl E.....ticket 297237, it's a part of the problem above, after that it only gets worse...

**Comment by KingKongst** [comment]
for problem above i have ticketnumber 300068

**Comment by Soolan** [comment]
Hello @KingKongst and @gre 
Thank you for raising this issue. I just wanted to mention I have the exact same problem.
I use Ubuntu 18.04 and my ticket number is: 313890
I will keep an eye on this issue and share if I found something.

**Comment by Soolan** [comment]
So I dig into the code a little and here the summary (and possibly the root cause of the problem).
But since my error reference is slightly different I put the screenshot here, hoping it might help someone

![Screenshot from 2019-06-01 13-49-19](https://user-images.githubusercontent.com/3588122/59767090-2439ca80-92d4-11e9-83a0-6bdbb40bf0b9.png)


First of all note there is a init() call in init.js (lines 48 and 52) which passes the collected user data to the helpers/db/index.js:

```db.init(userDataDirectory)```

But let's assume that user data (path) is collected properly and let's see what happens next.
Inside the helpers/db/index.js, there is a load function as follow:
```
**
 * Load a namespace, using <file>.json
 */
async function load(ns: string): Promise<mixed> {
  try {
    if (!DBPath) throw new NoDBPathGiven()
    const filePath = path.resolve(DBPath, `${ns}.json`)
    const fileContent = await fsReadFile(filePath)
    const { data } = JSON.parse(fileContent)
    memoryNamespaces[ns] = data
   // rest of the code

```
My error message clearly states an issue with unexpected end of JSON file which causes index.js fails at line 58:
`const filePath = path.resolve(DBPath, `${ns}.json`)`

Turned out that ${ns}.json is interpreted as "app.json" and when I do a search in my computer I can see it is under `~/.confgi/Ledger Live` folder (I'm using Ubuntu 18.04).

And when I open that file, I see the history of my transaction and at the end it seems the file is CUT OUT:
```{"data": {"user":{"id":"my-id"},"accounts":[{"data":{"id":"ripplejs:2:ripple: *** blah blah *** }, { *** more blah ***},{"id":"ripplejs:2:ripple:rapcpArsfFFVR8N```

You see? The last transaction is shown up to the first 15 characters of my wallet address. The rest of JSON file from that point onward  is cut out!

So having that said, the question is how can I recover the damaged **app.json** file.

P.S. As I mentioned in my ticket, nothing is missing and I can see the full history of my transactions on XRP Ledger.

Thanks in advance.

**Comment by gre** [comment]
We are testing on our side with Ubuntu 18.04 and can not reproduce the same issue, have you tried to reset the live and its data?

**Comment by Soolan** [comment]
Can you tell me how can I reset?

The reset button (Red one) does not do anything
The restart button, just restart the UI and leads to the same error message

**Comment by gre** [comment]
I think on Linux there is a `~/.config` folder. Open that folder and see if there is a LedgerLive folder in it. You can delete it and fully restore everything from the Ledger device.

**Comment by Soolan** [comment]
Thank you @gre 
That did the trick and I updated to v1.10 immediately

p.s. To everyone who read this thread.
In case you are using "Experimental USB" feature in linux Ubuntu 18.04, DEACTIVATE IT!
You will see the message during the update that say that. Pay attention to that message and it will save you from a lot of headaches.

Thanks again @gre

**Comment by gre** [comment]
We expect this to be fixed in context of a big rework that just got merged. we'll reopen in this does not fix in the next release #2596
